app/src/main/java/com/microsoft/graph/connect/GraphServiceClientManager.java

```diff
@@ -67,12 +77,19 @@ public synchronized IGraphServiceClient getGraphServiceClient() {
 
     public synchronized IGraphServiceClient getGraphServiceClient(IAuthenticationProvider authenticationProvider) {
         if (mGraphServiceClient == null) {
-            IClientConfig clientConfig = DefaultClientConfig.createWithAuthenticationProvider(
-                    authenticationProvider
-            );
-            mGraphServiceClient = new GraphServiceClient.Builder().fromConfig(clientConfig).buildClient();
-        }
+            MSALAuthenticationProvider msalAuthenticationProvider = new MSALAuthenticationProvider(
+                    getAppActivity(),
+                    GraphServiceClientManager.getApp(),
+                    mAuthenticationManager.getPublicClient(),
+                    Constants.SCOPES);
```

Comment:
Have msalAuthenticationProvider as a member variable and init only when its null. Same with graphClient.

Reply:
Done! Thanks